### ISSUE TITLE
Update quickstart guide to use Ubuntu 22.04

### ISSUE THREAD

**Issue by white238** [LLNL/serac#949]
We should remove the ubuntu 18/20 instructions and just go with this.

Also should include installing `lua5.4-dev` to avoid numerous Spack download and configure issues with curl, ca-certificates, and unzip that I ran into.

**Comment by white238** [comment]
`unzip` may be a better option to `apt-get` because `apt-get install lua5.4-dev` ends up in weird sub directories (`/usr/include/lua5.4/lua.h`).

**Comment by white238** [comment]
My mpi wrapper is injecting LTO flags which causes build failures from warnings in Axom. `cmake ... -DENABLE_WARNINGS_AS_ERRORS=OFF` can turn this off.

**Comment by samuelpmishLLNL** [comment]
In practice, the mpich compiler wrappers putting in LTO flags is more annoying than just creating warnings. It also can prevent you from using different compilers than the ones that were used to build the TPLs. In my personal notes for setting up a ubuntu 22.04 machine I have:

----------------------

# Disable LTO by default in MPICH w/ Ubuntu 22.04

by default, `sudo apt install`ing mpich will produce compiler wrappers that automatically insert 
`-flto` and other link-time optimization flags. This can produce binaries that are incompatible
with compilers other than the ones use to create those binaries. The solution is to manually
edit the compiler wrappers `/usr/bin/mpicc`, `/usr/bin/mpic++` to remove those flags. 

For example, change

```
final_ldflags=" -flto=auto -ffat-lto-objects -flto=auto -Wl,-Bsymbolic-functions -Wl,-z,relro"
```
to

```
final_ldflags=" -Wl,-Bsymbolic-functions -Wl,-z,relro"
```

--------------

Although it is kind of unpleasant to ask users to do this.

**Comment by white238** [comment]
I agree. I am not sure what the best path forward for this is.  Should we:

* Document what to remove from the wrapper and move on
* Remove known lto flags from the output of the MPI wrapper in our CMake
* Detect them and throw a warning/error for them to fix

**Comment by samuelpmishLLNL** [comment]
It might be worth contacting the ubuntu maintainer of mpich and asking for the lto flags be removed from future versions, but I'm not sure how well that would be received. 

I believe LTO is opt-in with gcc/clang, which makes it weird for mpich 4.0's mpicc to automatically enable it. I also don't know if it's possible to "opt-out" of LTO once the flags are in there (or how we would detect if those flags are present from CMake).

**Comment by white238** [comment]
Fixed in #1380
